src/approaches/probe_approach.py

```diff
@@ -0,0 +1,201 @@
+            sete: FrozenSet[GroundAtom]
+        ) -> Optional[Dict[FrozenSet[GroundAtom], _GroundNSRT]]:
+            if dictionary and sete in dictionary:
+                return dictionary[sete]
+            return None
+
+        def check_goal(atoms: FrozenSet[GroundAtom]) -> bool:
+            return goal.issubset(atoms)
+
+        def get_valid_actions(
+            atoms: FrozenSet[GroundAtom]
+        ) -> Iterator[Tuple[_GroundNSRT, float]]:
+            for op in get_applicable_operators(ground_nsrts, atoms):
+                yield (op, 1.0)
+
+        def get_next_state(atoms: FrozenSet[GroundAtom],
+                           ground_nsrt: _GroundNSRT) -> FrozenSet[GroundAtom]:
+            return frozenset(apply_operator(ground_nsrt, set(atoms)))
+
+        heuristic = create_task_planning_heuristic(
+            heuristic_name=CFG.pg3_task_planning_heuristic,
+            init_atoms=init,
+            goal=goal,
+            ground_ops=ground_nsrts,
+            predicates=self._initial_predicates,
+            objects=set(state),
+        )
+
+        _, action_seq = run_policy_guided_astar(
+            initial_state=initial_atoms,
+            check_goal=check_goal,
+            get_valid_actions=get_valid_actions,
+            get_next_state=get_next_state,
+            heuristic=heuristic,
+            policy=policy,
+            num_rollout_steps=CFG.pg3_max_policy_guided_rollout,
+            rollout_step_cost=0)
+
+        if action_seq is not None:
+            # If valid plan, add plan to memory so it can be refined!
+            memory["abstract_plan"] = action_seq
+            return memory["abstract_plan"].pop(0)
+        raise "Approach does not work"
+
+    def _generate_states(self, atoms: Set[GroundAtom], goal: Set[GroundAtom],
+                         state: State) -> Tuple[List, List]:
+        states = []
+        actions = []
+
+        initial_atoms = atoms
+
+        new_prompt = self._create_prompt(atoms, goal, [])
+        prompt = self._prompt_prefix + new_prompt
+        llm_predictions = self._llm.sample_completions(
+            prompt=prompt,
+            temperature=CFG.open_loop_llm_temperature,
+            seed=CFG.seed,
+            num_completions=5)
+
+        for llm_prediction in llm_predictions:
+            temp_states = []
+            temp_actions = []
+            temp_states.append(initial_atoms)
+            trimmed_prediction = ""
+            if len(llm_prediction) != 0 and llm_prediction[-1] != ')':
+                _, _, a = llm_prediction[::-1].partition(')')
+                trimmed_prediction = a[::-1] + ')'
+            else:
+                trimmed_prediction = llm_prediction
+
+            objects = set(state)
+            option_plan = self._llm_prediction_to_option_plan(
+                trimmed_prediction, objects)
+            if len(option_plan) == 0:
+                continue
+            nsrts = self._get_current_nsrts()
+            ground_nsrt_plan = []
```

Comment:
can you make this a separate helper method and add some comments explaining the assumption that operators and NSRTs are one and the same? (and that this would break for any non-PDDLEnv)